contracts/src/rollup/RollupCore.sol

```diff
@@ -592,7 +592,10 @@ abstract contract RollupCore is IRollupCore, PausableUpgradeable {
                 memoryFrame.sequencerBatchAcc,
                 wasmModuleRoot
             );
-            require(newNodeHash == expectedNodeHash, "UNEXPECTED_NODE_HASH");
+            require(
+                newNodeHash == expectedNodeHash || expectedNodeHash == bytes32(0),
```

Comment:
would be good for us to add a comment here that explains this check is only used for staker's not to have state change between their tx submission => inclusion (which can be opted out of by setting to `0`)

we can also remove the opt-out option in a future upgrade